fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -130,6 +131,99 @@ rocksdb::ReadOptions getReadOptions() {
 	return options;
 }
 
+struct ReadIterator {
+	uint64_t index;
+	bool inUse;
+	std::shared_ptr<rocksdb::Iterator> iter;
+	double creationTime;
+	ReadIterator(uint64_t index, DB& db, rocksdb::ReadOptions& options)
+	  : index(index), inUse(true), creationTime(now()), iter(db->NewIterator(options)) {}
+};
+
+class ReadIteratorPool {
+public:
+	ReadIteratorPool(DB& db) : db(db), index(0), numIteratorsReused(0), readRangeoptions(getReadOptions()) {
+		readRangeoptions.background_purge_on_iterator_cleanup = true;
+		readRangeoptions.auto_prefix_mode = (SERVER_KNOBS->ROCKSDB_PREFIX_LEN > 0);
+		TraceEvent(SevInfo, "ReadIteratorPool")
+		    .detail("KnobRocksDBReadRangeReuseIterators", SERVER_KNOBS->ROCKSDB_READ_RANGE_REUSE_ITERATORS);
+	}
+
+	void update() {
```

Comment:
Please comment on the behavior: what happens to the iterator already in use?
The caller is associated with commit action, so this seems to be "invalidating" existing iterators?